src/index.ts

```diff
@@ -1,8 +1,53 @@
-import type { TransformedSource, Transformer, TransformOptions } from '@jest/transform';
+import type { CacheKeyOptions, TransformedSource, Transformer, TransformOptions } from '@jest/transform';
 import type { Config } from '@jest/types';
+import createCacheKey from '@jest/create-cache-key-function';
 import { TsJestTransformer } from 'ts-jest/dist/ts-jest-transformer';
 
+import { NgJestConfig } from './config/ng-jest-config';
+import { stringify } from './utils/json';
+import { JsonableValue } from './utils/jsonable-value';
+
+interface ConfigSetItem {
+  configSet: NgJestConfig;
+  jestConfig: JsonableValue<Config.ProjectConfig>;
+}
+
 class AngularJestTransformer extends TsJestTransformer implements Transformer {
+  /**
+   * @internal
+   * cache config set between each test run
+   */
+  private static readonly configSetItems: ConfigSetItem[] = [];
+
+  /**
+   * Override `ts-jest` configsFor to initialize our `NgJestConfig`
+   * @internal
+   */
+  configsFor(jestConfig: Config.ProjectConfig): NgJestConfig {
+    let csi: ConfigSetItem | undefined = AngularJestTransformer.configSetItems.find(
+      (cs) => cs.jestConfig.value === jestConfig,
+    );
+    if (csi) return csi.configSet;
+    // try to look-it up by stringified version
+    const serialized = stringify(jestConfig);
+    csi = AngularJestTransformer.configSetItems.find((cs) => cs.jestConfig.serialized === serialized);
```

Comment:
We can avoid the upper `let` if we introduce a second variable for `csi`, like `const deserializedCsi = AngularJestTransformer.configSetItems...`

Reply:
Do you mean we don't need the `let csi` or do you mean 
```
const csi = AngularJestTransformer.configSetItems.find(
      (cs) => cs.jestConfig.value === jestConfig,
    );
if (csi) return csi.configSet;
// try to look-it up by stringified version
const serialized = stringify(jestConfig);
const deseralizedCsi = AngularJestTransformer.configSetItems...
```

Reply:
The second one :+1: